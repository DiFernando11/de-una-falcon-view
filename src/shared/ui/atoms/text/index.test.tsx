import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Text from '.';

describe('<Text />', () => {
  it('should match snapshot', () => {
    // Arrange
    const content = 'Texto snapshot';
    // Act
    const { asFragment } = render(
      <Text textType="heading" color="danger" as="h1">
        {content}
      </Text>
    );
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    // Arrange
    const content = 'Hola mundo';
    // Act
    render(<Text>{content}</Text>);
    // Assert
    expect(screen.getByText(content)).toBeInTheDocument();
  });

  it('should apply the primary color variant', () => {
    // Arrange
    const content = 'Texto azul';
    // Act
    render(<Text color="primary">{content}</Text>);
    const element = screen.getByText(content);
    // Assert
    expect(element).toHaveClass('text-blue-600');
  });
});
