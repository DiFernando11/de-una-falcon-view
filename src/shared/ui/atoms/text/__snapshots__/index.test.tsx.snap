// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Text /> should match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="font-family-space font-bold text-3xl leading-tight text-red-600"
  >
    Texto snapshot
  </h1>
</DocumentFragment>
`;
